src/main/java/com/easytestit/generatexml/GenerateXML.java

```diff
@@ -1,145 +1,151 @@
+     * <p>For example, if the configuration was transferred without creating a
+     * ZIP of the archive, then this functionality {@link com.easytestit.generatexml.service.ZipService()#createZip()}
+     * for creating the archive will not be launched.
+     * <p/>
+     *
      * When generating process fails report with information about error is provided.
-     * </p>
      */
     public void make() {
         //create new folder where new XML report will be created
         createResultsReportFolder();
 
         try {
-            if (configureXMLReport != null) {
-                //read JSON files from compiled directory
-                configureXMLReport.addJsonFiles(getJsonFilesFrom(configureXMLReport.getReportFolder()));
+            if (configuration != null) {
                 //convert JSON file in XML
-                new Serialization().serializeToXML(
-                        new TransformService().transformFeaturesToReport(
-                                new ParseJSON(configureXMLReport).parse()));
-                //create ZIP file from XML which created from previews step and send it to report server
-                if (configureXMLReport.containsConfigurationMode(ConfigureXMLMode.ZIP_XML_RESULT_TO_FILE)) createZip();
-                if (configureXMLReport.containsConfigurationMode(ConfigureXMLMode.SEND_RESULT_TO_RP)) sendXML();
+                convert(resolvePath(configuration.getSource()), configuration.getProjectName());
+                //create ZIP file from XML which created from previews step
+                boolean isZipCreated = false;
+                if (configuration.isReportAsZip()) {
+                    isZipCreated = configuration.getProjectName() != null
+                            ? new ZipService().createZip(configuration.getProjectName())
+                            : new ZipService().createZip(ConfigDataProvider.ZIP_NAME);
+                }
+
+                // send XML file to report server
+                if (configuration.isSendReport() && isZipCreated) {
+                    new SenderService().post(ConfigDataProvider.ZIP_NAME);
+                } else {
+                    LOGGER.log(Level.INFO, isZipCreated
+                            ? "Zip was created, but no need to send it"
+                            : "Zip was not created. Nothing to send to report server"
+                    );
+                }
+
             } else {
                 //when configuration is null start functionality with default parameters
-                new Serialization().serializeToXML(
-                        new TransformService().transformFeaturesToReport(
-                                new ParseJSON().parse(
-                                        getJsonFilesFrom(
-                                                new File(DefaultData.TARGET_FOLDER_PATH)))));
+                convert(resolvePath(new File("target/surefire-reports/")), "AggregatedReport");
```

Comment:
what if move "target/surefire-reports/" and "AggregatedReport" values to constants with meaningful name, like deafaultPathToReports, defaultProjectName

Reply:
I agree!

In addition, I added new functionality which takes from property default data of path to JSON's folder and of the project name. User could change these parameters and use a simple way of Generate-XML functionality.

Reply:
It will be available after fix where are changes in Travis CI file because for now, my commit was rejected dye to JavaDoc warnings and build was failed.